### PR TITLE
Test optional catch on generator throw

### PR DESCRIPTION
**Pull request by jridgewell**
It seems the current test is actually just testing that a normal throw will be caught by an optional-binding catch. I think it should be testing that resuming the generator with a throw will be properly caught, like how the `await` test is using a rejected promise's throw.

### CODE REVIEW

**Comment by ljharb** [comment]
Seems like if we're updating a test, we'd need to invalidate and recheck all results of that test.

**Comment by jridgewell** [comment]
How do I do that?

**Comment by ljharb** [comment]
The simplest is just to delete all the results for that test; the best but most work is to manually check the results of your updated test in each of the engines, and update the result accordingly.

**Comment by jridgewell** [comment]
I removed everything I wasn't able to verify.

**Review by ljharb — APPROVED** [review]

**Comment by ljharb** [comment]
@chicoxyzzy once you've stamped this, i'll rebase and merge it :-)

**Review by chicoxyzzy — COMMENTED** [review]
Test returns true in Safari 13.1. Also we have a [test runner for Ducktape](https://github.com/kangax/compat-table/blob/gh-pages/duktape.js).

**Comment by existentialism** [comment]
Test returns true for Safari 13.0.5 as well!

**Comment by jridgewell** [comment]
Tested in safari11_1 and duktape2_2, both remained the same.

File: data-es2016plus.js [hunk 1]
```diff
@@ -3013,15 +3012,12 @@ exports.tests = [
           firefox2: false,
           firefox57: false,
           firefox58: true,
-          opera10_50: false,
           chrome65: chrome.harmony,
           chrome66: true,
           safari11_1: true,
           safaritp: true,
           webkit: true,
           duktape2_2: false,
```
Comment (chicoxyzzy) [inline data-es2016plus.js:3020]: This should be `true` according to @jridgewell's [comment](https://github.com/kangax/compat-table/pull/1610#issuecomment-619465950)

Comment (jridgewell) [reply data-es2016plus.js:3020]: Sorry, I misspoke. I confirmed that this failed, as the original value did.

**Review by chicoxyzzy — APPROVED** [review]

**Pull request merged**